app.R

```diff
@@ -1,1104 +1,348 @@
+  )
+)
+
+server <- function(input, output,session) {
+  # Handle event when user selects dataset
+  observeEvent(input$type, {
+    itemPlot <<- "Protein"
+    if(input$type == "TMT") {
+      updateSelectizeInput(session, "genes", choices = TMT_choices$Gene, server = TRUE)
+      updateSelectizeInput(session, "gene", choices = TMT_choices$Gene, server = TRUE)
+      subtype <- c("Multiplot","Subtype", "Cytogenetics", "Fusion", "Mutations")
+      mutationLevels <<- c("WT", "Healthy Lin-")
+    }
+    else if(input$type == "LFQ") {
+      updateSelectizeInput(session, "genes", choices = LFQ_choices$Gene, server = TRUE)
+      updateSelectizeInput(session, "gene", choices = LFQ_choices$Gene, server = TRUE)
+      subtype <- c("Multiplot","Subtype", "Cytogenetics", "Fusion", "Mutations")
+      mutationLevels <<- c("WT","Healthy CD34", "Healthy Lin-")
+    }
+    else if(input$type == "Phosphosite") {
+      updateSelectizeInput(session, "gene", choices = phosphosite_choices$Gene, server = TRUE)
+      subtype <- c("Subtype", "Cytogenetics", "Fusion", "Mutations")
+      mutationLevels <<- c("WT", "Healthy Lin-")
+    }
+    else if(input$type == "mRNA") {
+      updateSelectizeInput(session, "genes", choices = mRNA_choices$Gene, server = TRUE)
+      updateSelectizeInput(session, "gene", choices = mRNA_choices$Gene, server = TRUE)
+      itemPlot <<- "Gene"
+      subtype <- c("Multiplot","Subtype", "Cytogenetics", "Fusion", "Mutations")
+      mutationLevels <<- c("WT","Healthy Donor CD19", "Healthy Donor CD3","Healthy Donor CD34","Healthy Donor Mono","Healthy Donor Neu","Healthy Donor Pro")
+    }
+    else if(input$type == "Protein vs mRNAs") {
+      updateSelectizeInput(session, "genes", label = "Enter Gene/Protein to plot", choices = genenames_corr$V1, server = TRUE)
+      shinyjs::show(id = "genes")
+      shinyjs::hide(id = "gene")
+      shinyjs::hide(id = "subtype")
+      subtype <- NULL
+    }
+    updateSelectizeInput(session, 'subtype', choices = subtype, server = TRUE)
+    if(input$type != "Protein vs mRNAs") {shinyjs::show(id = "subtype")}
+  })
+  
+  # Handle event when user selects subtype
+  observeEvent(input$subtype, {
+    # Phosphosite doesn't allow you to toggle clinical options
+    if(input$type == "Phosphosite") {
+      shinyjs::hide(id = "subtype_options")
+    }
+    else {
+      shinyjs::show(id = "subtype_options")
+    }
+    shinyjs::show(id = "gene")
+    shinyjs::hide(id = "genes")
+    shinyjs::hide(id = "mutation_status")
+    
+    if(input$subtype == "Multiplot") {
+      shinyjs::show(id = "genes")
+      shinyjs::hide(id = "subtype_options")
+      shinyjs::hide(id = "gene")
+      updateSelectizeInput(session, "genes", label = paste("Multi",tolower(itemPlot)," plot",sep=""))
+    }
+    else if(input$subtype == "Subtype") {
+      updateSelectizeInput(session, "gene", label = paste(itemPlot,"to plot"))
+      query <- paste0("SELECT DISTINCT FAB FROM ",input$type,"_Clinical;")
```

Comment:
nice - this is exactly the paradigm we want more of - having the app expose what's in the database without code changes